schema/atop_xslt.sch

```diff
@@ -13,63 +13,82 @@
     <!-- Global variables. -->
     <let name="docUri" value="document-uri(/)"/>
     
+    <let name="atopNamespaceUri" value="'http://www.tei-c.org/ns/atop'"/>
+    <let name="atopNamespacePrefix" value="'atop'"/>
+    
     <let name="reVarName" value="'^v[A-Z][a-zA-Z0-9]+(_[a-zA-Z0-9_]+)?$'"/>
     
     <let name="reParamName" value="'^t?p[A-Z][a-zA-Z0-9]+(_[a-zA-Z0-9_]+)?$'"/>
     
-    <let name="reFunctionName" value="'^teif:[a-z][a-z\-]+[a-z]$'"/>
+    <let name="reFunctionName" value="concat('^', $atopNamespacePrefix, ':[a-z][a-z\-]+[a-z]$')"/>
     
     <!-- Constraints -->
-    
-    <pattern id="things-must-have-as-attribute">
-        <rule context="xsl:template | xsl:variable | xsl:with-param | xsl:param | xsl:function">
-            <let name="ln" value="local-name(.)"/>
-            <let name="name" value="@name"/>
-            <assert test="@as">
-                ERROR: <value-of select="$name"/> (<value-of select="$ln"/>) missing required @as attribute.
+    <pattern id="names">
+        <title>Coding style rules for names</title>
+        <rule abstract="true" id="param-name-re">
+            <assert test="matches(tokenize(@name, ':')[last()], $reParamName)" id="assert-param-name-re" role="error">
```

Comment:
Hmmm… I would have used `substring-after( @name, ':' )` instead of `tokenize( @name, ':')[last()]`, on the theory that it is slightly easier to comprehend and probably a bit faster.

Reply:
(To be clear, I do not care which we use; our use case does not require serious optimization :-)

Reply:
```substring-after(@name, ':')``` returns the empty string if the name of the parameter has no prefix. ```tokenize(@name, ':')[last()]``` always returns the local name.

Reply:
Good point.